libbeat/tests/system/test_monitoring.py

```diff
@@ -45,10 +47,109 @@ def test_via_output_cluster(self):
+        self.wait_until(lambda: self.log_contains(re.compile("\[monitoring\].*Publish event")))
+        self.wait_until(lambda: self.log_contains(re.compile(
+            "Connection to .*elasticsearch\("+self.get_elasticsearch_monitoring_url()+"\).* established")))
+        self.wait_until(lambda: self.monitoring_doc_exists('beats_stats'))
+        self.wait_until(lambda: self.monitoring_doc_exists('beats_state'))
+
+        proc.check_kill_and_wait()
+
+        for monitoring_doc_type in ['beats_stats', 'beats_state']:
+            field_names = ['cluster_uuid', 'timestamp', 'interval_ms', 'type', monitoring_doc_type]
+            self.assert_monitoring_doc_contains_fields(monitoring_doc_type, field_names)
+
+    @unittest.skipUnless(INTEGRATION_TESTS, "integration test")
+    @attr('integration')
+    def test_compare(self):
+        """
+        Test that monitoring docs are the same, regardless of how they are shipped.
+        """
+
+        self.render_config_template(
+            "mockbeat",
+            xpack={
+                "monitoring": {
+                    "elasticsearch": {
+                        "hosts": [self.get_elasticsearch_url()]
+                    }
+                }
+            }
+        )
+
+        self.clean_output_cluster()
+        self.clean_monitoring_cluster()
+        self.init_output_cluster()
+
+        proc = self.start_beat(config="mockbeat.yml")
+        self.wait_until(lambda: self.log_contains("mockbeat start running."))
+        self.wait_until(lambda: self.log_contains(re.compile("\[monitoring\].*Publish event")))
+        self.wait_until(lambda: self.log_contains(re.compile(
+            "Connection to .*elasticsearch\("+self.get_elasticsearch_url()+"\).* established")))
+        self.wait_until(lambda: self.monitoring_doc_exists('beats_stats'))
+        self.wait_until(lambda: self.monitoring_doc_exists('beats_state'))
+
+        proc.check_kill_and_wait()
+
+        # TODO: Get a beats_stats doc
+        # TODO: Get a beats_state doc
+
+        self.render_config_template(
+            "mockbeat",
+            monitoring={
+                "elasticsearch": {
+                    "hosts": [self.get_elasticsearch_monitoring_url()]
+                }
+            }
+        )
+
+        self.clean_output_cluster()
+        self.clean_monitoring_cluster()
+
+        proc = self.start_beat(config="mockbeat.yml")
+        self.wait_until(lambda: self.log_contains("mockbeat start running."))
+        self.wait_until(lambda: self.log_contains(re.compile("\[monitoring\].*Publish event")))
+        self.wait_until(lambda: self.log_contains(re.compile(
+            "Connection to .*elasticsearch\("+self.get_elasticsearch_monitoring_url()+"\).* established")))
+        self.wait_until(lambda: self.monitoring_doc_exists('beats_stats'))
+        self.wait_until(lambda: self.monitoring_doc_exists('beats_state'))
+
+        proc.check_kill_and_wait()
+
+        # TODO: Get a beats_stats doc
```

Comment:
? leftovers or should stay?

Reply:
Worse, they actually need to be implemented in this PR. My mistake 🤦‍♂️. Implementing now...

Reply:
TODOs have been implemented now. Sorry about that!